src/Data/List/NonEmpty/Properties.agda

```diff
@@ -49,6 +58,46 @@ toList->>= : ∀ {ℓ} {A B : Set ℓ}
              (List⁺.toList (xs >>= f))
 toList->>= f (x ∷ xs) = begin
   List.concat (List.map (List⁺.toList ∘ f) (x ∷ xs))
-    ≡⟨ cong List.concat $ map-compose {g = List⁺.toList} (x ∷ xs) ⟩
+    ≡⟨ cong List.concat $ Listₚ.map-compose {g = List⁺.toList} (x ∷ xs) ⟩
   List.concat (List.map List⁺.toList (List.map f (x ∷ xs)))
     ∎
+
+length-++⁺ : (l : List A) (l′ : List⁺ A) → length (l ++⁺ l′) ≡ List.length l + length l′
+length-++⁺ [] l′          = refl
+length-++⁺ (x ∷ l) l′
+  rewrite length-++⁺ l l′ = refl
+
+length-++⁺-tail : (l : List A) (l′ : List⁺ A) → length (l ++⁺ l′) ≡ suc (List.length l + List.length (List⁺.tail l′))
+length-++⁺-tail [] l′          = refl
+length-++⁺-tail (x ∷ l) l′
+  rewrite length-++⁺-tail l l′ = refl
+
+++-++⁺ : (l : List A) → ∀ {l′ l″} → (l ++ l′) ++⁺ l″ ≡ l ++⁺ l′ ++⁺ l″
+++-++⁺ []      = refl
+++-++⁺ (x ∷ l) = cong (x ∷_) (cong toList (++-++⁺ l))
+
+++⁺-cancelˡ′ : ∀ l l′ {l″ l‴ : List⁺ A} → l ++⁺ l″ ≡ l′ ++⁺ l‴ → List.length l ≡ List.length l′ → l″ ≡ l‴
+++⁺-cancelˡ′ [] [] eq eql            = eq
+++⁺-cancelˡ′ (x ∷ l) (y ∷ l′) eq eql = ++⁺-cancelˡ′ l l′ (just-injective (cong fromList (cong List⁺.tail eq)))
+                                                         (suc-injective eql)
+
+++⁺-cancelˡ : ∀ l {l″ l‴ : List⁺ A} → l ++⁺ l″ ≡ l ++⁺ l‴ → l″ ≡ l‴
+++⁺-cancelˡ l eq = ++⁺-cancelˡ′ l l eq refl
+
+drop-++⁺ : ∀ (xs : List A) ys → drop (List.length xs) (xs ++⁺ ys) ≡ ys
+drop-++⁺ []       ys = refl
+drop-++⁺ (x ∷ xs) ys = drop-++⁺ xs ys
+
+map-++⁺-commute : ∀ (f : A → B) xs ys →
+                 map f (xs ++⁺ ys) ≡ List.map f xs ++⁺ map f ys
+map-++⁺-commute f [] ys       = refl
+map-++⁺-commute f (x ∷ xs) ys = cong (λ zs → f x ∷ toList zs) (map-++⁺-commute f xs ys)
+
+length-map : ∀ (f : A → B) xs → length (map f xs) ≡ length xs
+length-map f (_ ∷ xs) = cong suc (Listₚ.length-map f xs)
+
+map-cong : ∀ {f g : A → B} → f ≗ g → map f ≗ map g
+map-cong f≗g (x ∷ xs) = cong₂ _∷_ (f≗g x) (Listₚ.map-cong f≗g xs)
+
+map-compose : {g : B → C} {f : A → B} → map (g ∘ f) ≗ map g ∘ map f
```

Comment:
`f` and `g` should be explicit right? Can can we introduce them in alphabetical order?

Reply:
this property is directly taken from the list counterpart:
https://github.com/agda/agda-stdlib/blob/eec2bdf81779ba3e8c7a04e55398a3e3f4bf2e3a/src/Data/List/Properties.agda#L106-L108

Reply:
Hmm okay, well that really needs to be fixed too! Definitely swap it here, and I'll open an issue to swap the one in `Data.List.Properties`.

Reply:
also the name is actually a bit awkward. not sure why `compose` needs to be spelled out instead of `∘`

Reply:
Happy for that to be changed as well :+1: